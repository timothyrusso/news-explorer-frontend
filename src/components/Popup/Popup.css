.popup {
  position: fixed;
  /* Stay in place */
  z-index: 1;
  /* Sit on top */
  left: 0;
  /* Location of the box */
  top: 0;
  /* Location of the box */
  width: 100%;
  /* Full width */
  height: 100%;
  /* Full height */
  background-color: rgb(0, 0, 0);
  /* Fallback color */
  background-color: rgba(0, 0, 0, 0.5);
  /* Black w/ opacity */
  display: flex;
  justify-content: center;
  align-items: center;
  visibility: hidden;
  pointer-events: none;
  opacity: 0;
  transition: all linear 0.3s;
  backdrop-filter: blur(2px);
}

.popup_opened {
  visibility: visible;
  pointer-events: all;
  opacity: 1;
  transition-delay: 0s;
}

.popup__container {
  background-color: #fff;
  width: 100%;
  max-width: 430px;
  border-radius: 10px;
  position: relative;
}

.close-button {
  width: 100%;
  max-width: 40px;
  min-height: 40px;
  margin: auto;
  background-color: rgba(0, 0, 0, 0);
  border: none;
  position: absolute;
  top: -40px;
  right: -40px;
  z-index: 1;
  transition: opacity 0.2s linear;
  padding: 0;
  background-image: url(../../images/close-button.svg);
  background-repeat: no-repeat;
  background-position: center;
}

.close-button:hover {
  opacity: 0.6;
  cursor: pointer;
}

@media screen and (max-width: 552px) {
  .popup__container {
    max-width: 282px;
  }

  .close-button {
    max-width: 26px;
    min-height: 26px;
    top: -50px;
    right: 5px;
  }

  .close-button_type_info {
    top: -55px;
    right: 130px;
  }
}

@media screen and (max-width: 320px) {
  .popup__container {
    max-width: 320px;
    width: 100%;
    min-height: 512px;
    align-self: flex-end;
    border-radius: 10px 10px 0 0;
  }

  .close-button {
    max-width: 24px;
    min-height: 24px;
    top: -50px;
    right: 16px;
  }
}
