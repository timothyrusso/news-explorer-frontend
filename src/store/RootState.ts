import { UserState } from './user/user.reducer';
import { ArticleState } from './article/article.reducer';
import { TogglesState } from './toggles/toggles.reducer';
import { ErrorsState } from './errors/errors.reducer';

export type RootState = {
  user: UserState;
  article: ArticleState;
  toggles: TogglesState;
  errors: ErrorsState;
};
