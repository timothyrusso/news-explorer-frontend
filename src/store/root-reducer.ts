import { combineReducers } from 'redux';
import { userReducer } from './user/user.reducer';
import { articleReducer } from './article/article.reducer';

export const rootReducer = combineReducers({
  user: userReducer,
  article: articleReducer,
});
