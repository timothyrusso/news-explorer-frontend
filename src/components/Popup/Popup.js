import { useEffect } from 'react';
import { usePopup } from '../../hooks/usePopup';
import './Popup.css';

const Popup = ({ isOpen, name, infoTooltip, children }) => {
  const { closeAllPopups } = usePopup();
  // here is `useEffect` for the `Escape` listener
  useEffect(() => {
    // with this we prevent adding the listener if the popup is not opened
    if (!isOpen) return;
    // we should define the handler inside `useEffect`, so that it wouldn’t lose the reference to be able to remove it
    const closeByEscape = (evt) => {
      if (evt.key === 'Escape') {
        closeAllPopups();
      }
    };

    document.addEventListener('keydown', closeByEscape);
    // don’t forget to remove the listener in the `clean-up` function
    return () => document.removeEventListener('keydown', closeByEscape);
    // here we watch `isOpen` to add the listener only when it’s opened
  }, [isOpen, closeAllPopups]);

  // here is the overlay handler
  const handleOverlay = (evt) => {
    if (evt.target === evt.currentTarget) {
      closeAllPopups();
    }
  };

  // then we add the main wrapper with class `popup` and `popup_opened`
  return (
    <div
      className={`popup ${isOpen ? 'popup_opened' : ''} popup_type_${name}`}
      onClick={handleOverlay}
    >
      {/* the container for the contents */}
      <div className="popup__container">
        {/* here will be anything you add as `children`*/}
        {children}
        {/* add the close button */}
        <button
          aria-label="Close"
          type="button"
          className={`close-button ${
            infoTooltip ? 'close-button_type_info' : ''
          }`}
          onClick={closeAllPopups}
        ></button>
      </div>
    </div>
  );
};

export default Popup;
