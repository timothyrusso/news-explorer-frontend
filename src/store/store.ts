import { legacy_createStore as createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import { rootReducer } from './root-reducer';
import { ThunkDispatch } from 'redux-thunk';
import { AnyAction } from 'redux';
import { RootState } from './RootState';

const store = createStore(rootReducer, applyMiddleware(thunk));

export type AppDispatch = ThunkDispatch<RootState, unknown, AnyAction>;

export default store;
