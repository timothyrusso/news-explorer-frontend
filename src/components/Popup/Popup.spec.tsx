import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../../store/store';
import Popup from './Popup';

const mockChild = (
  <>
    <h2 className="popup__title">Mock Title</h2>
    <div>Actual Content Here</div>
    <button
      type="submit"
      className="submit-button popup__button"
      aria-label="Mock Submit"
    >
      Mock Button Text
    </button>
    <div>Mocked FormRedirect Component</div>
  </>
);

describe('Popup component', () => {
  it('Should render the Popup component', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <Router>
          <Popup isOpen={true} name="name" infoTooltip={false}>
            {mockChild}
          </Popup>
        </Router>
      </Provider>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
