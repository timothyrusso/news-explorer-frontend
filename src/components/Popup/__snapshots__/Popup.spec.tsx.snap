// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popup component Should render the Popup component 1`] = `
<DocumentFragment>
  <div
    class="popup popup_opened popup_type_name"
  >
    <div
      class="popup__container"
    >
      <h2
        class="popup__title"
      >
        Mock Title
      </h2>
      <div>
        Actual Content Here
      </div>
      <button
        aria-label="Mock Submit"
        class="submit-button popup__button"
        type="submit"
      >
        Mock Button Text
      </button>
      <div>
        Mocked FormRedirect Component
      </div>
      <button
        aria-label="Close name popup"
        class="close-button "
        type="button"
      />
    </div>
  </div>
</DocumentFragment>
`;
